src/helper.js

```diff
@@ -0,0 +1,63 @@
+export const mapDataIntoLap = (inputLaps, lapKey, data) => {
+  const laps = JSON.parse(JSON.stringify(inputLaps));
+  let index = 0;
+  for (let i = 0; i < laps.length; i++) {
+    const lap = laps[i];
+    const nextLap = laps[i + 1];
+    const tempData = [];
+    for (let j = index; j < data.length; j++) {
+      const row = data[j];
+      if (nextLap) {
+        if (lap.start_time <= row.timestamp.toISOString() && nextLap.start_time > row.timestamp.toISOString()) {
```

Comment:
Is `lap.start_time` an ISO String format? If so, maybe it would be better to convert it to timestamp and compare numbers. Comparing numbers (lower/higher) fells more natural (to me) than strings.

Reply:
Agree, I changed the start_time, timestamp all into timestamp.